app/models/user.rb

```diff
@@ -91,8 +91,13 @@ def setting_auto_play_gif
     settings.auto_play_gif
   end
 
-  def activate_session
-    session_activations.activate(SecureRandom.hex).session_id
+  def activate_session(request)
+    detection = Browser.new(request.user_agent)
+
+    session_activations.activate(session_id: SecureRandom.hex,
```

Comment:
@akihikodaki platform.name returns an english string like "Generic Linux", while id returns something like "linux", "windows", so it's easier to localize. I thought that browser.name would not contain english strings since most browsers have trademark names like firefox/internet explorer which do not need to be translated, but if as you say it also returns "Generic Browser", then I need to switch that to id too